lang/cs/Org.Apache.REEF.Network/Group/Driver/Impl/GroupCommDriver.cs

```diff
@@ -147,6 +148,24 @@ public ICommunicationGroupDriver NewCommunicationGroup(string groupName, int num
         }
 
         /// <summary>
+        /// Remove a group from the GroupCommDriver
+        /// Throw ArgumentException if the group does not exist
+        /// </summary>
+        /// <param name="groupName"></param>
+        public void RemoveCommunicationGroup(string groupName)
+        {
+            lock (_groupsLock)
+            {
+                if (!_commGroups.ContainsKey(groupName))
+                {
+                    Exceptions.Throw(new ArgumentException("Group Name is not registered with GroupCommunicationDriver"), LOGGER);
```

Comment:
I added Org.Apache.REEF.Utilities.Diagnostics.Exceptions couple of years ago. Our decision was to use it everywhere in REEF code so that we can control the messages in a central place. I actually replaced REEF code consistently at that time. But later people start to use .Net throw method directly again.